pkgs/misc/vscode-extensions/python/default.nix

```diff
@@ -14,16 +14,48 @@ assert ctagsUseFixed -> null != ctags;
 let
   pythonDefaultsTo = if pythonUseFixed then "${python}/bin/python" else "python";
   ctagsDefaultsTo = if ctagsUseFixed then "${ctags}/bin/ctags" else "ctags";
-in
 
-vscode-utils.buildVscodeMarketplaceExtension {
+  # The arch tag comes from 'PlatformName' defined here:
+  # https://github.com/Microsoft/vscode-python/blob/master/src/client/activation/types.ts
+  arch =
+    if stdenv.isLinux && stdenv.isx86_64 then "linux-x64"
+    else if stdenv.isDarwin then "osx-x64"
+    else throw "Only x86_64 Linux and Darwin are supported.";
+
+  languageServerSha256 = {
+    "linux-x64" = "0mqjl3l1zk1zd7n0rrb2vdsrx6czhl4irdm4j5jishg9zp03gkkd";
+    "osx-x64" = "1csq8q8fszv9xk9qiabg12zybxnzn8y2jsnvjrlg4b8kvm63sz40";
+  }."${arch}";
+
+  # version is languageServerVersion in the package.json
+  languageServer = extractNuGet rec {
```

Comment:
I believe what was meant by @worldofpeace in his review of the preceding incarnation of this PR was that it would be preferable that `languageServer` be moved in its own file (as it can be used by other packages) and taken as input.

This would indeed remove most of the complexity as the added build inputs are to my understanding meant only for the download, extraction and automated patchelf-ing of the language server:

`stdenv, fetchurl, extractNuGet, icu, curl, openssl, lttng-ust, autoPatchelfHook`

The `extractNuGet` utility could then be moved out of of the extension sub folder as well to be potentially used by others.

This is however merely a recommendation as it could indeed be done later as the need arises.